components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m

```diff
@@ -447,6 +466,46 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
   [self.view setNeedsLayout];
 }
 
+- (void)expandToFullScreenWithDuration:(NSTimeInterval)duration {
+  CGFloat topSafeArea = 0;
+  if (@available(iOS 11.0, *)) {
+    topSafeArea = self.view.safeAreaInsets.top;
+  }
+  CGFloat totalHeight = self.headerViewController.preferredContentSize.height +
+      self.contentViewController.preferredContentSize.height;
+  CGFloat precentageOfFullScreen = totalHeight / CGRectGetHeight(self.presentingViewBounds);
+  if (CGRectGetHeight(self.presentingViewBounds) > totalHeight) {
+    CGFloat maxViewHeight = CGRectGetHeight(self.presentingViewBounds) - totalHeight;
```

Comment:
can you move up properties that are mutual to both the "if" and "else" blocks, i see some similarity.